custom_components/echonetlite/climate.py

```diff
@@ -42,17 +43,17 @@ async def async_setup_entry(hass, config_entry, async_add_devices):
     entities = []
     for entity in hass.data[DOMAIN][config_entry.entry_id]:
         if entity['instance_data']['eojgc'] == 1 and entity['instance_data']['eojcc'] == 48: #Home Air Conditioner
-             entities.append(EchonetClimate(config_entry.data["title"], entity['API'], TEMP_CELSIUS))
-    async_add_devices(entities)
+             entities.append(EchonetClimate(config_entry.data["title"], entity['API'], hass.config.units))
+    async_add_devices(entities, True)
```

Comment:
This ensures that update is called before the entity is 'available' so it always has valid data